### PR TITLE
Encoding the circuit text drawer (based on preferred encoding) and ignore errors.

### LINKED ISSUE #2616: Text Drawer fails on windows/appveyor
### Information

- **Qiskit Terra version**: master (but also effects releases)
- **Python version**: Any
- **Operating system**: Windows

### What is the current behavior?

When a test or script prints a circuit using the text drawer as part of CI on windows/appveyor  testing it always raises a UnicodeEncodeError. For example something like:

```
    Traceback (most recent call last):'
     File "C:\\projects\\qiskit-sdk-py\\test\\python\\test_examples.py", line 50, in test_all_examples'
       self.assertEqual(run_example.returncode, 0, error_string)'
     File "C:\\Python37\\lib\\unittest\\case.py", line 839, in assertEqual'
       assertion_func(first, second, msg=msg)'
     File "C:\\Python37\\lib\\unittest\\case.py", line 832, in _baseAssertEqual'
       raise self.failureException(msg)'
    AssertionError: 1 != 0 : Running example circuit_draw.py failed with return code 1'
    'stdout:'
    'stderr: Traceback (most recent call last):\r'
      File "C:\\projects\\qiskit-sdk-py\\examples\\python\\circuit_draw.py", line 34, in <module>\r'
        print(bell_circuit)\r'
     File "C:\\Python37\\lib\\encodings\\cp1252.py", line 19, in encode\r'
       return codecs.charmap_encode(input,self.errors,encoding_table)[0]\r'
    UnicodeEncodeError: 'charmap' codec can't encode characters in position 8-12: character maps to <undefined>\r"
```

Most recently: https://ci.appveyor.com/project/Qiskit/qiskit-sdk-py/builds/25126953/job/9tmvcy2iutrh9qp9#L2063

### Steps to reproduce the problem

Print a circuit to the console during an appveyor job.

### What is the expected behavior?

The circuit prints successfully.

### Suggested solutions

Figure out why this happens in the CI environment and fix it.

### PR DESCRIPTION
**Pull request by 1ucian0**
Fixes #2616

This allows to run on Windows when the *reasonable*  encoding available. Otherwise, ignore those errors (the test on windows run with cp1252, a very limited charset).

### CODE REVIEW

**Comment by 1ucian0** [comment]
The current CI error might be related to https://github.com/mtreinish/stestr/issues/295

File: test/python/test_examples.py [hunk 1]
```diff
@@ -32,7 +32,6 @@
 class TestPythonExamples(QiskitTestCase):
     """Test example scripts"""
 
-    @unittest.skipIf(os.name == 'nt', 'Skip on windows until #2616 is fixed')
```
Comment (Cryoris) [inline test/python/test_examples.py:35]: Should this test still be skipped if only the limited charset is available? Or should we force Windows users to use a more general charset in that case? 🤔

Comment (1ucian0) [reply test/python/test_examples.py:33]: I hacked the way around https://github.com/Qiskit/qiskit-terra/pull/4900/commits/695bb4759dcf3180eb34def8def3b83754bc3c51

File: test/python/test_examples.py [hunk 2]
```diff
@@ -44,15 +43,15 @@ def test_all_examples(self):
...
                 stdout, stderr = run_example.communicate()
                 error_string = "Running example %s failed with return code %s\n" % (
                     example, run_example.returncode)
                 error_string += "stdout:%s\nstderr: %s" % (
-                    stdout.decode('utf8'), stderr.decode('utf8'))
+                    stdout, stderr)
                 self.assertEqual(run_example.returncode, 0, error_string)
 
-    @unittest.skipIf(os.name == 'nt', 'Skip on windows until #2616 is fixed')
     @unittest.skipIf(sys.platform == 'darwin' and sys.version_info[1] == 8,
```
Comment (mtreinish) [inline test/python/test_examples.py:55]: Don't we need to set the encoding in this test too?

Comment (1ucian0) [reply test/python/test_examples.py:55]: Indeed. Fixed in 449a3d5

**Comment by 1ucian0** [comment]
>  also do you think we need to have a release note for the new encoding option?

Most of the user, I assume, use the drawer as `print(circuit)`, and I'm not exposing the option to `draw` and only to dunderscore methods. So.. not for now?

**Pull request merged**